lib/ContractInteract/GnosisSafe.js

```diff
@@ -0,0 +1,383 @@
+   *
+   * @param {String} owner New owner address.
+   * @param {String} threshold New threshold.
+   *
+   * @returns {String} Executable data for adding owner with threshold.
+   */
+  getAddOwnerWithThresholdExecutableData(owner, threshold) {
+    return this.contract.methods.addOwnerWithThreshold(owner, threshold).encodeABI();
+  }
+
+  /**
+   * Allows to remove an owner and update the threshold at the same time.
+   *
+   * @param {String} prevOwner Owner that pointed to the owner to be removed in the linked list.
+   * @param {String} owner Owner address to be removed.
+   * @param {String} threshold New threshold.
+   *
+   * @returns {String} Executable data for removing an owner with threshold.
+   */
+  getRemoveOwnerWithThresholdExecutableData(prevOwner, owner, threshold) {
+    return this.contract.methods.removeOwner(prevOwner, owner, threshold).encodeABI();
+  }
+
+  /**
+   * Allows to swap/replace an owner with another address.
+   *
+   * @param {String} prevOwner Owner that pointed to the owner to be replaced in the linked list.
+   * @param {String} oldOwner Owner address to be replaced.
+   * @param {String} newOwner New owner address.
+   *
+   * @returns {String} Executable data for swapping an owner.
+   */
+  getSwapOwnerExecutableData(prevOwner, oldOwner, newOwner) {
+    return this.contract.methods.swapOwner(prevOwner, oldOwner, newOwner).encodeABI();
+  }
+
+  /**
+   * Allows to update the number of required confirmations by Safe owners.
+   * This can only be done via a Safe transaction.
+   *
+   * @param {String} threshold New threshold.
+   *
+   * @returns {String} Executable data to change threshold.
+   */
+  getChangeThresholdExecutableData(threshold) {
+    return this.contract.methods.changeThreshold(threshold).encodeABI();
+  }
+
+  /**
+   * Allows to execute a Safe transaction confirmed by required number of owners and then pays the account that
+   * submitted the transaction.
+   *
+   * @param {String} to Destination address of Safe transaction.
+   * @param {String} value Ether value of Safe transaction.
+   * @param {String} data Data payload of Safe transaction.
+   * @param {Number} operation Operation type of Safe transaction.
+   * @param {String} safeTxGas Gas that should be used for the Safe transaction.
+   * @param {String} dataGas Gas costs for data used to trigger the safe transaction and to pay the payment transfer
+   * @param {String} gasPrice Gas price that should be used for the payment calculation.
+   * @param {String} gasToken Token address (or 0 if ETH) that is used for the payment.
+   * @param {String} refundReceiver Address of receiver of gas payment (or 0 if tx.origin).
+   * @param {String} signatures Packed signature data ({bytes32 r}{bytes32 s}{uint8 v}).
+   *
+   * @returns {Object} Raw transaction object.
+   */
+  execTransactionRawTx(to, value, data, operation, safeTxGas, dataGas, gasPrice, gasToken, refundReceiver, signatures) {
+    if (!Web3.utils.isAddress(to)) {
```

Comment:
I am not sure `to` is the only mandatory params here 🤔

Reply:
data is also mandatory. Added that as validation 👍